### ISSUE TITLE
Feature Add support of os.chflags() on Linux platform

### ISSUE THREAD

**Issue by 01e27b45-90f2-4c74-9e5e-7e7e54c3d78e** [python/cpython#75494]
BPO | [31313](https://bugs.python.org/issue31313)
--- | :---
Nosy | @pitrou, @larryhastings, @socketpair
Files | <li>[chiflag.c](https://bugs.python.org/file47115/chiflag.c "Uploaded as text/x-csrc at 2017-09-01.09:09:24 by @socketpair")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-08-31.09:55:01.045>
labels = ['3.7', 'type-feature', 'library']
title = 'Feature Add support of os.chflags() on Linux platform'
updated_at = <Date 2017-09-01.09:27:54.508>
user = 'https://github.com/socketpair'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-01.09:27:54.508>
actor = 'pitrou'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2017-08-31.09:55:01.045>
creator = 'socketpair'
dependencies = []
files = ['47115']
hgrepos = []
issue_num = 31313
keywords = []
message_count = 5.0
messages = ['301043', '301046', '301089', '301091', '301092']
nosy_count = 4.0
nosy_names = ['pitrou', 'larry', 'neologix', 'socketpair']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue31313'
versions = ['Python 3.7']
```

</p></details>

**Comment by 01e27b45-90f2-4c74-9e5e-7e7e54c3d78e** [comment]
Yes, chflags() is not supported by linux directly. But actually, Linux supports chflags functionality:

https://stackoverflow.com/questions/34176464/why-os-chflags-doesnt-work-under-linux

through

ioctl (fd, EXT2_IOC_SETFLAGS, \<BITMASK_OF_FLAGS\>)

**Comment by pitrou** [comment]
Judging by the ioctl name, this is ext2/ext3/ext4 specific?

**Comment by 01e27b45-90f2-4c74-9e5e-7e7e54c3d78e** [comment]
Well, it seems, you are right.

But there are another ioctls (maybe they are the same, I haven't checked):

FS_IOC_SETFLAGS
FS_IOC_GETFLAGS

http://man7.org/tlpi/code/online/book/files/chiflag.c.html  (also attached)

**Comment by pitrou** [comment]
On Ubuntu 16.04 I get the following values:
EXT2_IOC_SETFLAGS = 0x40086602, FS_IOC_GETFLAGS = 0x80086601

**Comment by pitrou** [comment]
Uh, of course, we care about SETFLAGS here.  Which gives:
EXT2_IOC_SETFLAGS = 0x40086602, FS_IOC_SETFLAGS = 0x40086602

**Comment by encukou** [comment]
`FS_IOC` flags are different from the `stat.UF_*` &  `stat.SF_*` that `os.chflags` operates on.
If we add them, they should use a different function, perhaps in `fcntl` or around `statx`, or use the `chattr` name.

Feature proposals should go on Discourse nowadays; ideally with a use case.
